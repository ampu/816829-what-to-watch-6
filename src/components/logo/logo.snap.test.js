import React from 'react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';

import {render} from '@testing-library/react';

import Logo from './logo';

describe(`Logo component`, () => {
  it(`Should render according to the snapshot`, () => {
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <Logo/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot when isLight`, () => {
    const {container} = render((
      <Router history={createMemoryHistory()}>
        <Logo isLight/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot when user navigates away from '/' url`, () => {
    const history = createMemoryHistory();
    history.push(`/login`);

    const {container} = render((
      <Router history={history}>
        <Logo/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });
});
