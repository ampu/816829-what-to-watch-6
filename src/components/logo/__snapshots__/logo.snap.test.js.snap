// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component Should render according to the snapshot 1`] = `
<div>
  <div
    class="logo"
  >
    <a
      aria-current="page"
      class="logo__link logo__link--active"
      href="/"
    >
      <span
        class="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        class="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        class="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
</div>
`;

exports[`Logo component Should render according to the snapshot when isLight 1`] = `
<div>
  <div
    class="logo"
  >
    <a
      aria-current="page"
      class="logo__link logo__link--light logo__link--active"
      href="/"
    >
      <span
        class="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        class="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        class="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
</div>
`;

exports[`Logo component Should render according to the snapshot when user navigates away from '/' url 1`] = `
<div>
  <div
    class="logo"
  >
    <a
      class="logo__link"
      href="/"
    >
      <span
        class="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        class="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        class="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
</div>
`;
